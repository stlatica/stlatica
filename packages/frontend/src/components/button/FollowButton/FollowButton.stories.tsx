import React from "react";
import type { Story, StoryDefault } from "@ladle/react";
import { FollowButton } from "./FollowButton";
// import FollowButton from "./FollowButton";

export default {
  title: "components/button/FollowButton",
} satisfies StoryDefault;

export const Story1: Story = () => {
  return <FollowButton initialFollowState={false} />;
};
