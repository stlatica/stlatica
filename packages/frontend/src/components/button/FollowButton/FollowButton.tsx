import React from "react";

import { CancelButton } from "@/components/button/CancelButton";
import { SubmitButton } from "@/components/button/SubmitButton";

type FollowButtonProps = {
  readonly isFollow: boolean;
  readonly onClick?: () => void;
};

export const FollowButton: React.FC<FollowButtonProps> = ({ isFollow, onClick }) => {
  if (isFollow) {
    return (
      <CancelButton className="w-[8em]" onClick={onClick}>
        フォロー中d
      </CancelButton>
    );
  }

  return (
    <SubmitButton className="w-[8em]" onClick={onClick}>
      フォロー
    </SubmitButton>
  );
};
