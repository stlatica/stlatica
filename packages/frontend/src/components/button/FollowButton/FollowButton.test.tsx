import React from "react";
import { render } from "@testing-library/react";
import { describe, expect, test } from "vitest";
import { Story1, Story2, Story3 } from "./FollowButton.stories";

describe("snapshot", () => {
  test("snapshot1", () => {
    const { asFragment } = render(<Story1 />);
    expect(asFragment()).toMatchSnapshot();
  });
  test("snapshot2", () => {
    const { asFragment } = render(<Story2 />);
    expect(asFragment()).toMatchSnapshot();
  });
  test("snapshot3", () => {
    const { asFragment } = render(<Story3 />);
    expect(asFragment()).toMatchSnapshot();
  });
});
