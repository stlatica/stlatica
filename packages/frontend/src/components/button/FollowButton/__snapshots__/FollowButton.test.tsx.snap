// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snapshot > snapshot1 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
    style="width: 8em;"
    tabindex="0"
    type="button"
  >
    フォロー
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;

exports[`snapshot > snapshot2 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
    style="width: 8em;"
    tabindex="0"
    type="button"
  >
    フォロー中
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;

exports[`snapshot > snapshot3 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
    style="width: 8em;"
    tabindex="0"
    type="button"
  >
    フォロー
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
